identity/v3userpass.go

```diff
@@ -103,14 +103,16 @@ func (u *V3UserPass) Auth(creds *Credentials) (*AuthDetails, error) {
 			},
 		},
 	}
-	auth.Auth.Scope = &v3AuthScope{
-		Project: &v3AuthProject{
-			Domain: &v3AuthDomain{
-				Name: projectDomain,
+	if creds.TenantName != "" || creds.TenantID != "" {
```

Comment:
Even if both id and name are "", we still need to pass in the project domain I would think? Maybe that's not the case?

Reply:
As discussed, This conditional check is required.
Refer : https://docs.openstack.org/keystone/pike/_modules/keystone/auth/controllers.html

Reply:
Need a test for this

Reply:
What happens if neither is set?